agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/heartbeat/DefaultHeartBeatPropertyProvider.java

```diff
@@ -103,7 +105,9 @@ public Boolean call() {
             }
           } catch (RuntimeException e) {
             if (logger.isWarnEnabled()) {
-              logger.warn("Failed to obtain heartbeat property", e);
+              try (MDC.MDCCloseable ignored = AzureMonitorMsgId.HEARTBEAT_SEND_ERROR.makeActive()) {
+                logger.warn("Failed to obtain heartbeat property", e);
+              }
```

Comment:
have we seen this message in the diagnostics logs? I'm not sure I really understand the purpose of this catch block in the first place